Cython/Compiler/Nodes.py

```diff
@@ -704,6 +704,19 @@ def analyse(self, return_type, env, nonempty=0, directive_locals=None, visibilit
 
         exc_val = None
         exc_check = 0
+
+        if (env.directives["legacy_implicit_noexcept"]
+                and not return_type.is_pyobject
```

Comment:
I think we should include also memoryviews. But this should be done in future PR adding warnings for `noexcept` functions returning memoryviews.

Reply:
I don't know about memoryviews, sorry.

Thanks for approving the PR. It'd be great to have a released version of cython with this PR merged, to help with fixing sagemath missing `noexcept` so we can eventually remove `legacy_implicit_noexcept` for good.